localstack-core/localstack/utils/container_utils/container_client.py

```diff
@@ -1183,6 +1189,20 @@ def parse_additional_flags(
                 hosts_split = add_host.split(":")
                 extra_hosts[hosts_split[0]] = hosts_split[1]
 
+        # set env file values before env values, as the latter override the earlier
+        if args.env_files:
+            env_vars = env_vars if env_vars is not None else {}
+            for env_file in args.env_files:
+                with open(env_file, mode="rt") as f:
+                    env_file_lines = f.readlines()
```

Comment:
Adding a guard here to check if the file actually exists and logging a warning with some context/instructions (e.g. that the file needs to be in the container) would be helpful here!

Reply:
Added a log error for FileNotFoundError and OSError with some hints. I still reraised the exception, I think the container creation should probably fail if the `DOCKER_FLAGS` are incorrect.